update_branch_protections.py

```diff
@@ -0,0 +1,234 @@
+        for row in csv_reader:
+            data_row = ProtectionGroup.from_dict(row)
+            repo_data.append(data_row)
+    return repo_data
+
+
+# This could be a decorator, but we only do this once here
+parse_repo_list.__doc__ = parse_repo_list.__doc__.format(
+    '\n      - '.join(list(ProtectionGroup.CHECK_NAME_MAP.keys())),
+    '\n    - '.join(ProtectionGroup.RULE_MAP.keys())
+)
+
+
+def main(
+    owner: str = "pcdshub",
+    repo_name: str = "",
+    repo_type: str = "Python Library",
+    prot_master: bool = False,
+    prot_pages: bool = False,
+    prot_default: bool = False,
+    repo_data_path: str = "",
+    write: bool = False
+):
+    if repo_data_path:
+        # run for each repo in list
+        repo_data = parse_repo_list(repo_data_path)
+        for protection_group in repo_data:
+            protection_group.apply_protections(write=write)
+        return
+
+    if not repo_name:
+        return
+
+    # apply settings for a single repo
+    protection_group = ProtectionGroup(
+        owner=owner, repo_name=repo_name, repo_type=repo_type,
+        master=prot_master, gh_pages=prot_pages, default=prot_default
+    )
+    protection_group.apply_protections(write=write)
+
+    return
+
+
+def _create_argparser() -> argparse.ArgumentParser:
+    """
+    Create an ArgumentParser for update_branch_protections
+
+    Returns
+    -------
+    argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser()
+    # specify an owner, repo, and settings
+    parser.add_argument("owner", type=str, default='pcdshub', nargs='?',
+                        help='Organization or owner of the repository, "pcdshub"'
+                             'by default')
+    parser.add_argument("repo_name", type=str, default='', nargs='?',
+                        help='Name of the repository')
+    parser.add_argument("repo_type", type=str, default='Other', nargs='?',
+                        help='Type of the repository.  Should be one of '
+                             f'{list(ProtectionGroup.CHECK_NAME_MAP.keys())}')
+    parser.add_argument("--protect-master", action="store_true", dest='prot_master',
+                        help="Add master protection.  This should be applied as "
+                             "much as possible. This rule requires pull requests and "
+                             "disallows force pushes. This protection also adds required "
+                             "status checks depending on the repo_type")
+    parser.add_argument("--protect-pages", action="store_true", dest='prot_pages',
+                        help="Add gh-pages branch protection, allowing force pushes "
+                             "to the gh-pages branch specifically")
+    parser.add_argument("--protect-default", action="store_true", dest='prot_default',
+                        help="Add default (*) branch protection, disallows "
+                             "the creation of new branches")
```

Comment:
I know we've agreed to the forking life, but I still expect this to cause some chaos. See the `mfx` repo, for example.

Reply:
`mfx` - out of our purview - should probably be set to `exempt`?

Reply:
possibly, now that it's being managed by the beamline staff themselves

Reply:
I think as those who have been burdened with maintaining these packages, we also reserve the right to impose our conventions on libraries.  I hope that having well defined workflows with proper guardrails makes contributing easier to understand for newer users.  (I've been meaning to continue writing some more Pull Request disambiguation docs)

I do agree that we can move the specific repos to `exempt` (nnn)  or `other` (ynn, for master protections) as needed.  If anything I hope this tool makes it easier to do so quickly.